Docs: Add authentication note for external/internal image sources (#83469)
vercel/next.js: docs/01-app/03-api-reference/02-components/image.mdx

}
```

> > **Good to know**: When specifying an external URL or API Routes (internal URL) as the image source, the image may be processed through Next.js built-in [Image Optimization](/docs/app/getting-started/images).

Reviewer: ```suggestion
> Good to know: When Next.js Image Optimization is used, for security reasons, request headers are not forwarded to the image source. If retrieving the image requires authentication, consider using the [unoptimized](#unoptimized) property to disable Image Optimization.
```

or if we still want to mention the "internal URL" caveat:

```suggestion
> Good to know: When Next.js Image Optimization is used, request headers are not forwarded to the image source, even when that source is your own Next.js server (API Routes or assets from public folder). If retrieving the image requires authentication, consider using the [unoptimized](#unoptimized) property to disable Image Optimization.
```

Also the image optimization link was not pointing at relevant docs

> For security reasons, when Image Optimization is used, request headers are not forwarded to API Routes or external URLs.
> If retrieving the image data requires authentication, consider using the [unoptimized](#unoptimized) property to disable Image Optimization.

> 

Reviewer: ```suggestion
> Good to know: When Next.js Image Optimization is used, for security reasons, request headers are not forwarded to the image source. If retrieving the image requires authentication, consider using the [unoptimized](#unoptimized) property to disable Image Optimization.
```

or if we still want to mention the "internal URL" caveat:

```suggestion
> Good to know: When Next.js Image Optimization is used, request headers are not forwarded to the image source, even when that source is your own Next.js server (API Routes or assets from public folder). If retrieving the image requires authentication, consider using the [unoptimized](#unoptimized) property to disable Image Optimization.
```

Also the image optimization link was not pointing at relevant docs

Author: @icyJoseph Thank you for your feedback and fix.
Since the intention is to make the restrictions and behavior clearer to developers, I prefer the more detailed latter change.

}
```

> > Good to know: When Next.js Image Optimization is used, request headers are not forwarded to the image source, even when that source is your own Next.js server (API Routes or assets from public folder). If retrieving the image requires authentication, consider using the [unoptimized](#unoptimized) property to disable Image Optimization.

Reviewer 2: ```suggestion
```

> #### `alt`

Reviewer 2: ```suggestion
> **Good to know**: For security reasons, the Image Optimization API using the default [loader](#loader) will _not_ forward headers when fetching the `src` image.
```
